script.js

```diff
@@ -0,0 +1,165 @@
+// Global Elements
+let scoreGrid = [['','',''] ,['','',''], ['','','']];
+let playerOneTurn = true;
+let bot = false;
+
+// Dom Elements
+const colorMode = document.getElementById('color-mode');
+const gameMode = document.getElementById('game-mode');
+const gameGrids = document.querySelectorAll('.grid-space');
+const message = document.getElementById('message');
+const gridTable = document.querySelector('.grid-table');
+
+
+let currentColor =  'dark';
+
+
+const insertInScoreGrid = (current,pos,symb) => {
+    current.innerText = symb;
+    const [one, two] = get2DPosition(pos);
+    scoreGrid[one][two] = symb;
+}   
+
+
+const get2DPosition = (pos) => {
+    let counter = 0;
+    for (let i = 0; i < scoreGrid.length; i++) {
+        for (let k = 0; k < scoreGrid[i].length; k++) {
+            if (counter === pos && scoreGrid[i][k] === '') return [i,k]; 
+            counter++;
+        }
+    }
+}
+
+
+
+// Check Grid
+const checkWin = (symb) => { 
+    
+    // Check Columns and Rows
+    for (let rows = 0; rows < scoreGrid.length; rows++) {
+        const currentColumn = [];
+        const currentRow = [];
+        for (let cols = 0; cols < scoreGrid[rows].length; cols++) {
+            currentColumn.push(scoreGrid[cols][rows]);
+            currentRow.push(scoreGrid[rows][cols]);
+        }   
+        if (currentColumn.every(val => val === symb) ||  currentRow.every(val => val === symb) ) return true;
+    }
+
+    // Check Diagonals
+   const checkDiag = (scoreGrid[0][0] === symb && scoreGrid[1][1] === symb && scoreGrid[2][2] === symb) || (scoreGrid[0][2] === symb && scoreGrid[1][1] === symb && scoreGrid[2][0] === symb) 
+
+   if (checkDiag) return true
+    
+    return false;
+}
+
+
+const checkFilled = () => {
+    return [...gameGrids].every(grid => grid.innerText !== '');
```

Comment:
It looks like you're trying to use the spread operator to flatten the array, but I don't think it actually does that? If I'm wrong and it does flatten it, this isn't typically what spread is used for and it's less clear to read than Array.prototype.flat. If I'm missing something that the spread operator is doing something clever here, please let me know!